src/uu/ls/src/ls.rs

```diff
@@ -2027,45 +2212,12 @@ fn display_file_name(
         }
     }
 
-    if config.format == Format::Long && path.file_type()?.is_symlink() {
-        if let Ok(target) = path.p_buf.read_link() {
-            name.push_str(" -> ");
-
-            // We might as well color the symlink output after the arrow.
-            // This makes extra system calls, but provides important information that
-            // people run `ls -l --color` are very interested in.
-            if let Some(ls_colors) = &config.color {
-                // We get the absolute path to be able to construct PathData with valid Metadata.
-                // This is because relative symlinks will fail to get_metadata.
-                let mut absolute_target = target.clone();
-                if target.is_relative() {
-                    if let Some(parent) = path.p_buf.parent() {
-                        absolute_target = parent.join(absolute_target);
-                    }
-                }
-
-                let target_data = PathData::new(absolute_target, None, None, config, false);
-
-                // If we have a symlink to a valid file, we use the metadata of said file.
-                // Because we use an absolute path, we can assume this is guaranteed to exist.
-                // Otherwise, we use path.md(), which will guarantee we color to the same
-                // color of non-existent symlinks according to style_for_path_with_metadata.
-                let target_metadata = match target_data.md() {
-                    Some(md) => md,
-                    None => path.md()?,
-                };
-
-                name.push_str(&color_name(
-                    ls_colors,
-                    &target_data.p_buf,
-                    target.to_string_lossy().into_owned(),
-                    target_metadata,
-                ));
-            } else {
-                // If no coloring is required, we just use target as is.
-                name.push_str(&target.to_string_lossy());
-            }
-        }
+    if config.format == Format::Long
+        && path.file_type().is_some()
+        && path.file_type().unwrap().is_symlink()
+    {
+        name.push_str(" -> ");
+        name.push_str(&path.p_buf.read_link().unwrap().to_string_lossy());
```

Comment:
This is a tricky one: should we stick we GNU or do something that's a little bit better? I think the original comment for this decision makes a good point: if I have a symlink to a certain file, I think it's valuable to color that to the filetype of the file being linked to. If you have a different view, I'd be happy to hear it, but preferably in an issue or another PR :)

Reply:
I'm sorry.  You're right this was insensitive on my part without explaining the change.  My view is slightly different in that, yes, this is helpful feature, but I suppose my greatest annoyance with uutils is when the defaults don't match GNU behavior exactly, including the error messages and order.  It makes debugging more difficult.  In the colors scenario, this is less important, but requires working around for uutils tests and perhaps for GNU tests.  However we are also getting crushed in performance for metadata call sensitive ops, of which this seems like another.

Maybe another colors= option to opt in?

Reply:
> It makes debugging more difficult.

Debugging the utils or debugging scripts using the utils? This particular difference seems harmless to me, but if you open an issue we can maybe get some more opinions on this.

Reply:
Sure.  You're right.  Deserves more discussion.   I'll keep the test designed for this commented out with a note.